src/lib/capacityEngine.js

```diff
@@ -32,78 +32,115 @@ function getUtilizationRate(capacityInAssets, capacityUsedNXM) {
+ * @returns {BigNumber} The available capacity as a BigNumber.
+ */
+function calculateAvailableCapacity(trancheCapacities, allocations, firstUsableTrancheIndex) {
+  const unused = trancheCapacities.reduce((available, capacity, index) => {
+    const allocationDifference = capacity.sub(allocations[index]);
+    const allocationToAdd =
+      index < firstUsableTrancheIndex
+        ? bnMin(allocationDifference, Zero) // only carry over the negative
+        : allocationDifference;
+    return available.add(allocationToAdd);
+  }, Zero);
+  return bnMax(unused, Zero);
+}
+
 /**
  * Calculates capacity and pricing data for a specific tranche of product pools.
  *
  * @param {Array<Object>} productPools - Array of product pool objects.
  * @param {number} firstUsableTrancheIndex - Index of the first usable tranche.
  * @param {boolean} useFixedPrice - Flag indicating whether to use fixed pricing.
  * @param {BigNumber} now - Current timestamp in seconds.
- * @returns {Object} An object containing capacity used, capacity available, minimum price, and total premium.
+ * @param {Object} assets - Object containing asset information.
+ * @param {Object} assetRates - Object containing asset rates.
+ * @returns {Object} An object containing aggregated data and capacity per pool.
  */
-function calculateProductDataForTranche(productPools, firstUsableTrancheIndex, useFixedPrice, now) {
-  return productPools.reduce(
-    (accumulated, pool) => {
-      const { capacityUsedNXM, capacityAvailableNXM, minPrice, totalPremium } = accumulated;
-      const { allocations, trancheCapacities, targetPrice, bumpedPrice, bumpedPriceUpdateTime } = pool;
-
-      // calculating the capacity in allocation points
-      const used = allocations.reduce((total, allocation) => total.add(allocation), Zero);
-      const total = trancheCapacities.reduce((total, capacity) => total.add(capacity), Zero);
-
-      const unused = trancheCapacities.reduce((available, capacity, index) => {
-        const allocationDifference = capacity.sub(allocations[index]);
-        return index < firstUsableTrancheIndex
-          ? available.add(bnMin(allocationDifference, Zero)) // only carry over the negative
-          : available.add(allocationDifference);
-      }, Zero);
-
-      const availableCapacity = bnMax(unused, Zero);
-
-      // convert to nxm
-      const totalInNXM = total.mul(NXM_PER_ALLOCATION_UNIT);
-      const usedInNxm = used.mul(NXM_PER_ALLOCATION_UNIT);
-      const availableInNXM = availableCapacity.mul(NXM_PER_ALLOCATION_UNIT);
-
-      if (availableCapacity.isZero()) {
-        // only add up the used capacity and return the same values for the rest
-        return {
-          capacityUsedNXM: usedInNxm.add(capacityUsedNXM),
-          capacityAvailableNXM,
-          minPrice,
-          totalPremium,
-        };
-      }
+function calculateProductDataForTranche(productPools, firstUsableTrancheIndex, useFixedPrice, now, assets, assetRates) {
```

Comment:
I would remove `assets` and `assetRates` from this function, as I can see we are not using `availableCapacityInAssets` for anything and we are recalculating it again later in the code

Reply:
the `availableCapacityInAssets` in this function is under `capacityPerPool` field, meaning the value is segregated by StakingPool.

on the other hand the 2nd `capacityInAssets` (on capacityEngine function) is the value across all StakingPools.